### ISSUE TITLE
How do I react to a click on a marker?

### ISSUE THREAD

**Issue by MrYossu** [rungwiroon/BlazorGoogleMaps#128]
Just trying to get to grips with this excellent package, but finding the documentation a little sparse, and the same code a little hard to follow.

Suppose I have a collection of customers in my database, and want to show a marker for each one. That's no problem, the [map marker sample](https://github.com/rungwiroon/BlazorGoogleMaps/blob/master/ServerSideDemo/Pages/MapMarker.razor.cs#L59) shows it fine.

However, what I actually want to is that when the user clicks on a marker, I don't want the little pop-up window to show, I want a Blazor event calling, so I can show more details in a separate part of the page.

So, I have two questions...

1. How do I wire up the click event of the marker? The `click` event shown in the sample code is raised when you click on the map, not on a marker.

2. How do I know which marker was clicked? Ideally Id like the database `Id` of the customer.

In case it makes any difference, I intend to use clustering.

Thanks for any help you can give.

**Comment by valentasm1** [comment]
You need to search in issues for "Marker click". 
Also i agree that documentation is worst. Server side has most updated samples.
More info about marker click event
https://github.com/rungwiroon/BlazorGoogleMaps/issues/118

**Comment by valentasm1** [comment]
```
private async Task AddMarker()
        {
            var mapCenter = await map1.InteropObject.GetCenter();

            var marker = await Marker.CreateAsync(map1.JsRuntime, new MarkerOptions()
            {
                Position = mapCenter,
                Map = map1.InteropObject,
                Label = $"Test {markers.Count}",
                //Animation = Animation.Bounce
                //Icon = new Icon()
                //{
                //    Url = "https://developers.google.com/maps/documentation/javascript/examples/full/images/beachflag.png"
                //}
                //Icon = "https://developers.google.com/maps/documentation/javascript/examples/full/images/beachflag.png"
            });

            markers.Push(marker);

            return;
            await bounds.Extend(mapCenter);

            var icon = await marker.GetIcon();

            Console.WriteLine($"Get icon result type is : {icon.Value.GetType()}");

            icon.Switch(
                s => Console.WriteLine(s),
                i => Console.WriteLine(i.Url),
                _ => { });

            markers.Push(marker);

            await marker.AddListener<MouseEvent>("click", async e =>
            {
                var markerLabel = await marker.GetLabel();
                _events.Add("click on " + markerLabel);
                StateHasChanged();

                await e.Stop();
            });
        }
```

**Comment by MrYossu** [comment]
@valentasm1 Thanks for the replies, but the code you showed (whcih I spotted in the sample just after I posted) only gives you the label. What if I have two companies with the same name?

I need to get at something that identifies the company uniquely, and the name doesn't do that.

Thanks again.

**Comment by valentasm1** [comment]
You have MapMarker.Guid on top of that you could create your own solution.
I think you expect smth like windows forms have Tag property which could be anything. IMO i think i done that in my solution. You need extend MapMarker and add custom property

**Comment by MrYossu** [comment]
Great thanks, that was what I needed.
